### PR TITLE
[Unit] OperatorDistributor.processNextMinipool()

### PR DESCRIPTION
**Pull request by huy-nodeset**

### CODE REVIEW

File: test/MockContracts/MockWEth.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (huy-nodeset) [reply test/MockContracts/MockWEth.sol:1]: Sounds good, let's come back to this later

File: test/unit/Constellation/OperatorDistributor/test-processNextMinipool.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                    });
+                });
+
+                describe("when the minipool is managed by Constellation", function () {
+                    beforeEach(async function () {
+                        await expect(mockSuperNode.setIsMinipoolRecognized(mockMinipool.address, true)).to.not.be.reverted;
+                    });
+
+                    describe("when the minipool is finalised", function () {
+                        it("does nothing", async function () {
```
[the remark this answers is not in the document]
Comment (huy-nodeset) [reply test/unit/Constellation/OperatorDistributor/test-processNextMinipool.ts:159]: added or already there

File: test/unit/Constellation/OperatorDistributor/test-calculateRplStakeShortfall.ts [hunk 3]
```diff
@@ -39,7 +37,7 @@ describe("OperatorDistributor.calculateRplStakeShortfall", function () {
         expect(await operatorDistributor.minimumStakeRatio()).equals(ethers.utils.parseEther("0.15"));
     });
 
-    describe("test case 1", async function () {
+    describe("test case 1", function () {
```
Comment (VVander) [inline test/unit/Constellation/OperatorDistributor/test-calculateRplStakeShortfall.ts:40]: you don't want these parallelized?

File: test/unit/Constellation/OperatorDistributor/test-onEthBeaconRewardsReceived.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                    expect(await ethers.provider.getBalance(treasury.address)).to.equal(ethers.BigNumber.from("0"));
+                    expect(await ethers.provider.getBalance(operatorReward.address)).to.equal(ethers.BigNumber.from("0"));
+
+                    await expect(operatorDistributor.connect(owner).onEthBeaconRewardsReceived(
+                        ethers.utils.parseEther("10"),
+                        ethers.utils.parseEther("0.1"),
+                        ethers.utils.parseEther("0.2"),
+                    )).to.not.be.rejected;
+
+                    expect(await ethers.provider.getBalance(treasury.address)).to.equal(ethers.BigNumber.from("1000000000000000000"));
```
Comment (VVander) [inline test/unit/Constellation/OperatorDistributor/test-onEthBeaconRewardsReceived.ts:146]: FYI you can use the following for something a little more readable
```suggestion
                    expect(await ethers.provider.getBalance(treasury.address)).to.equal(ethers.utils.parseEther("1"));
```

**Review by VVander — DISMISSED** [review]
Approved with comments

**Review by VVander — APPROVED** [review]

File: test/unit/Constellation/OperatorDistributor/test-calculateRplStakeShortfall.ts (hunk 3 above)
Comment (huy-nodeset) [reply test/unit/Constellation/OperatorDistributor/test-calculateRplStakeShortfall.ts:40]: will do later. I wanted more describe names but just called them 1-4 for their entries in the spreadsheet for now

**Pull request merged**